### PR TITLE
[Telemetry Operator] Harden otel-collector deployment

### PR DESCRIPTION
**Pull request by skhalash**
**Description**

Changes proposed in this pull request:

- Add collector liveness/readiness probes
- Set collector liveness/readiness probes according to https://github.com/open-telemetry/opentelemetry-operator/issues/1264
- Split collector service into an OTLP and an OpenCensus service
- Make CPU/memory limits configurable

**Related issue(s)**
https://github.com/kyma-project/kyma/issues/16198

### CODE REVIEW

File: components/telemetry-operator/main.go [hunk 1]
```diff
@@ -140,9 +146,13 @@ func main() {
...
-	flag.StringVar(&traceCollectorDeploymentName, "trace-collector-deployment-name", "telemetry-trace-collector", "Deployment name for tracing OpenTelemetry Collector")
+	flag.BoolVar(&traceCollectorCreateServiceMonitor, "trace-collector-create-service-monitor", true, "Create Prometheus ServiceMonitor for opentelemetry-collector")
+	flag.StringVar(&traceCollectorBaseName, "trace-collector-base-name", "telemetry-trace-collector", "Default name for tracing OpenTelemetry Collector Kubernetes resources")
+	flag.StringVar(&traceCollectorOTLPServiceName, "trace-collector-otlp-service-name", "telemetry-otlp-traces", "Default name for tracing OpenTelemetry Collector Kubernetes resources")
 	flag.StringVar(&traceCollectorImage, "trace-collector-image", otelImage, "Image for tracing OpenTelemetry Collector")
+	flag.StringVar(&traceCollectorPriorityClass, "trace-collector-priority-class", "", "Priority class name for tracing OpenTelemetry Collector")
+	flag.StringVar(&traceCollectorCPULimit, "trace-collector-cpu-limit", "1", "CPU limit for tracing OpenTelemetry Collector")
```
Comment (chrkl) [inline components/telemetry-operator/main.go:154]: Maybe we should add those settings also to the values.yaml file to have a chance to adjust later without a new release.

Should we have the reservations also configurable?

**Review by chrkl — COMMENTED** [review]
Looks very good, only a small suggestion. We should also adjust the [trace-demo](https://github.com/kyma-project/examples/tree/main/trace-demo) example to the new service names after merging.

File: components/telemetry-operator/main.go (hunk 1 above)
Comment (skhalash) [reply components/telemetry-operator/main.go:154]: Yeah, makes sense, will do it.

**Review by chrkl — APPROVED** [review]

**Review by chrkl — APPROVED** [review]

**Pull request merged**